src/components/screens/ImportSecretKeyScreen.tsx

```diff
@@ -54,10 +57,28 @@ const ImportSecretKeyScreen: React.FC<ImportSecretKeyScreenProps> = ({
     }
 
     try {
-      await importSecretKey(secretKey);
-      navigation.goBack();
+      await importSecretKey(secretKey, password);
+
+      navigation.dispatch(
+        CommonActions.reset({
+          index: 0,
+          routes: [
+            {
+              name: ROOT_NAVIGATOR_ROUTES.MAIN_TAB_STACK,
+              state: {
+                index: 1,
+                routes: [
+                  { name: MAIN_TAB_ROUTES.TAB_HISTORY },
+                  { name: MAIN_TAB_ROUTES.TAB_HOME },
+                  { name: MAIN_TAB_ROUTES.TAB_DISCOVERY },
+                ],
+              },
+            },
+          ],
+        }),
+      );
```

Comment:
On VerifyPasswordScreen we have something similar but cleaner, have we tried using the same here?:
```
        navigation.reset({
          index: 0,
          // @ts-expect-error: This is a valid route.
          routes: [{ name: ROOT_NAVIGATOR_ROUTES.MAIN_TAB_STACK }],
        });
```